### PR TITLE
add wait option to "helm uninstall" and "helm install"

### PR DESCRIPTION
**Pull request by muffl0n**

### CODE REVIEW

File: src/main/java/io/oneko/kubernetes/impl/DeploymentManagerImpl.java [hunk 1]
```diff
@@ -110,6 +110,8 @@ public ReadableProjectVersion deploy(final WritableProjectVersion version) {
 					helmCommands.uninstall(deployment.getReleaseNames());
 				}
 
+				Thread.sleep(10000);
```
Comment (philmtd) [inline src/main/java/io/oneko/kubernetes/impl/DeploymentManagerImpl.java:113]: Won't this also block the HTTP request sent by the frontend?

And shouldn't this be in the block above, so it's only executed when an application is actually uninstalled?

**Review by philmtd — CHANGES_REQUESTED** [review]
I'm not fully convinced yet that this is the best solution for the problem.

As far as I see what we want to achieve is: The uninstallation needs to be completed before we install the application again.

We provide our own Helm API wrapper in the `helmapi` package. Do you think we could add the `--wait` parameter of `helm delete` (and other commands that offer it) optionally and exposed through a method parameter, so we let Helm wait until all resources are deleted? Could this work?

**Comment by muffl0n** [comment]
>  ... so we let Helm wait until all resources are deleted? Could this work?

Sounds good! I thought that this was already done. I'm expecting a race condition between the deletion of the ingress, the ingress controller reacting to it and the creation of the new ingress.
So waiting explicitly for the deletion sounds like a wise idea.

**Comment by muffl0n** [comment]
Added `--wait` to `helm uninstall`. Didn't make it optional, cause there is no sense in not waiting, imho.

**Comment by philmtd** [comment]
I'd like to have the `--wait` argument optionally on both the `uninstall` and `install` methods. That way we could prevent HTTP calls coming from the API to have to wait for the uninstallation to finish. Waiting is not required either when the scheduled shutdown is configured. We literally only require to wait when we immediately after that plan to install again. So I'd really love to have that argument and to be able to pass it optionally (default should be `false`).

**Review by philmtd — COMMENTED** [review]
I made the change I wanted. Could you review this please @muffl0n?

**Comment by muffl0n** [comment]
LGTM! I can't approve cause it's my PR. :)

**Review by philmtd — APPROVED** [review]

**Pull request merged**